website/identifiers/utils.py

```diff
@@ -94,19 +94,24 @@ def request_identifiers_from_ezid(target_object):
 
         client = get_ezid_client()
         already_exists = False
+        new = True
         try:
             resp = client.create_identifier(doi, metadata)
         except HTTPError as error:
             already_exists = True
             if 'identifier already exists' not in error.message.lower():
                 raise
             resp = client.get_identifier(doi)
-
+            new = False
+        if new:
+            resp = {'success': resp['success'].split('|')[0]}
```

Comment:
Instead of parsing the response manually twice, it might be better to leave the response alone here in the `request_identifiers_from_ezid` method, and do the parsing later in `parse_identifiers` below.  Parsing only once leaves less room for error and would be easier to catch if something changes with the response in the future

Reply:
Ideally yes, but this is used by two different functions for preprints and nodes.